### ISSUE TITLE
unserialize(): Error at offset 

### ISSUE THREAD

**Issue by zhengwhizz** [thomasjohnkane/snooze#52]
in laravel version "version": "v7.14.1", i got this error, i found databse notification field data is not right,
`O:41:"App\Notifications\AppointmentNotification":10:{s:11:"`
it's missing something. Why?
i have a Appointment model to store user's appoinment, 
and how can i get the notification by Appointment id? if user change the appointment time, i need to reschedule the notification, i only can get the notification by user id / notification type

**Comment by atymic** [comment]
Is it possible that you renamed or moved the class?

**Comment by zFly** [comment]
Hello,

I experienced exactly the same error, and fixed it using the `fix-null-bytes` branch from this [pull request](https://github.com/thomasjohnkane/snooze/pull/55/files).

Writing `serialize(new PaymentReminderNotification($this->order));` returns the expected serialized string

```
"O:45:"App\Notifications\PaymentReminderNotification":9:{s:8:"\x00*\x00order";O:45:"Illuminate\Contracts\Database\ModelIdentifier":4:{s:5:"class";s:16:"App\Models\Order";s:2:"id";s:36:"bfac1542-4669-3822-89a5-ec2aa1cd7add";s:9:"relations";a:0:{}s:10:"connection";s:5:"pgsql";}s:2:"id";N;s:6:"locale";N;s:10:"connection";N;s:5:"queue";N;s:15:"chainConnection";N;s:10:"chainQueue";N;s:5:"delay";N;s:7:"chained";a:0:{}}"
```

But after storing it into my pgsql DB the string gets cut of

```
"O:45:\"App\\Notifications\\PaymentReminderNotification\":9:{s:8:\""
```

Are there any plans to support pgsql DBs or merge the pull request?

**Comment by korridor** [comment]
I am having the same problem as @zFly. 

I researched the problem a bit and I think the problem is that the response of `serialize()` should not be written to a database field with a text type.
A quote from the [PHP documentation of serialize](https://www.php.net/manual/en/function.serialize.php):

>  Note that this is a binary string which may include null bytes, and needs to be stored and handled as such. For example, serialize() output should generally be stored in a BLOB field in a database, rather than a CHAR or TEXT field. 

According to [this comment](https://www.php.net/manual/de/function.serialize.php#96504) this is only a problem if you use `private` or `protected` properties.

**Comment by Onatcer** [comment]
Laravel itself [sets all the properties accessible while serializing models](https://github.com/laravel/framework/blob/b0fb1ba07a248bc63a092b608746f681db4d54f2/src/Illuminate/Queue/SerializesModels.php#L129). As far as i can tell that's the reason they don't experience any problems like this when storing jobs. Would you accept a PR implementing a similar method or using the Laravel Trait?

**Comment by atymic** [comment]
Could we use a `blob` field to avoid this issue?

**Comment by korridor** [comment]
@atymic I think so, but I think the advantage of saving the serialized object to `TEXT` is that it is semi-readable.
Maybe something like this would work: https://github.com/illuminate/cache/blob/7aacad834d7472b19a590f4f2fc0e413cf0efd83/DatabaseStore.php#L356

**Comment by atymic** [comment]
Closed in #64 
And will release in v2 :)
